Add org.entangle_photo.Manager

manuq opened the pull request:
I'm trying to contribute [Entangle](https://entangle-photo.org) to flathub.

I proposed this upstream in the mailing list: https://groups.google.com/forum/#!topic/entangle-devel/R3-n_5nrVV8

Current status: it builds, the app launches, and the UI is displayed. But I couldn't yet connect my Canon camera, which works in Entangle (same app version) installed from Fedora packages. I tried passing `--device=all` but still doesn't work. Hopefully someone can help me to debug this.

All dependencies are borrowed from other apps already in flathub:

- libghoto2: from [Shotwell](https://github.com/flathub/org.gnome.Shotwell/blob/86f43c9f8943181b860d34576825ae2005e7dd62/org.gnome.Shotwell.json)

- gudev (which in turn requires systemd for udev): from [org.mixxx.Mixxx](https://github.com/flathub/org.mixxx.Mixxx/blob/c2c4c4cc0f0d9178d410457e3f725f22ac6fdb94/org.mixxx.Mixxx.yaml)

- libpeas: from [gedit](https://github.com/flathub/org.gnome.gedit/blob/41d61cce73789703b1468da13b065d3bf6eff0b9/org.gnome.gedit.json)

- exiv2 (required by gexiv2 but Shotwell had it from git and I preferred an archive): from [Photos](https://github.com/flathub/org.gnome.Photos/blob/5b8c8ffc9b3650c5efa6ac98875320be3dc62287/org.gnome.Photos.json)

- gexiv2 (which in turn requires exiv2): from [Shotwell](https://github.com/flathub/org.gnome.Shotwell/blob/86f43c9f8943181b860d34576825ae2005e7dd62/org.gnome.Shotwell.json)

- libraw (and libraw-pkgconfig.patch): from [Photos](https://github.com/flathub/org.gnome.Photos/blob/5b8c8ffc9b3650c5efa6ac98875320be3dc62287/org.gnome.Photos.json)

Also I verified that the desktop file pass validation, and I have patched the appstream file to validate. I also psoposed this patch upstream here: https://gitlab.com/entangle/entangle/merge_requests/6

bilelmoussaoui commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,153 @@
+{
+    "app-id": "org.entangle_photo.Manager",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.32",
```
Bump to 3.34 if possible.

TingPing replied:
And remove DConf permissions once done.

bilelmoussaoui commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,153 @@
+    "app-id": "org.entangle_photo.Manager",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.32",
+    "branch": "stable",
```
```suggestion
```

manuq replied:
Does this mean that flathub calls `flatpak-builder` with `--default-branch=stable` ? Just to know so I can update my command to test locally.

manuq replied:
The App Requirements page answered, yes. https://github.com/flathub/flathub/wiki/App-Requirements#branches-and-runtimes

bilelmoussaoui commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,153 @@
+    "branch": "stable",
+    "sdk": "org.gnome.Sdk",
+    "command": "entangle",
+    "tags": [],
```
```suggestion
```

bilelmoussaoui commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,153 @@
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options": {
```
Please remove those empty tags.

TingPing commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,153 @@
+            "config-opts" : ["--disable-introspection", "--disable-docs"]
+        },
+        {
+            "name": "systemd",
```
Is this for udev? You can remove once bumped to 3.34 runtime.

manuq commented:
Thanks for the reviews! I pushed changes, all except the bump to 3.34 because I cannot test it right now.The fetch of libgphoto2 is temporarily? failing with `datapacket.dl.sourceforge.net port 443: No route to host`. But from my personal notes, last time I tried with 3.34 I got this build error:
```
./configure: line 8778: intltool-update: command not found
checking for intltool >= 0.40.0...  found
configure: error: Your intltool is too old.  You need intltool 0.40.0 or later.
```

manuq commented:
I went to Sourceforge and found a different URL for libgphoto2. So I updated it in the manifest. I will propose the same change for Shotwell.

manuq commented:
`intltool-update` is not in the SDK 3.34 (it is in 3.32), so module `libpeas` fails to build if I bump the runtime. Should I build `intltool`?

manuq commented:
All comments addressed!

bilelmoussaoui commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,131 @@
+            ]
+        },
+        {
+            "name": "intltool",
```
To avoid duplicates, please use intltool from shared-modules.

manuq replied:
Is not needed anymore.

bilelmoussaoui commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,131 @@
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.gnome.org/sources/libpeas/1.22/libpeas-1.22.0.tar.xz",
```
Bump to 1.24. I don't think you will need intltool after that

bilelmoussaoui commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,131 @@
+        {
+            "name": "entangle",
+            "buildsystem": "meson",
+            "config-opts": [],
```
```suggestion
```

bilelmoussaoui commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,131 @@
+                    "sha256" : "d8af23364aa40fd8607f7e073df74e7ace05582f4ba13f1724d12d3c97e8852d"
+                }
+            ],
+            "config-opts" : ["--disable-introspection", "--disable-docs"]
```
Nitpick: put config-opts before sources.

bilelmoussaoui commented:
bot, build org.entangle_photo.Manager

manuq commented:
I pushed renames for the icons, desktop file and appdata file.

manuq commented:
bot, build org.entangle_photo.Manager

manuq commented:
OK it builds now. Still I can't make it work with my SLR camera. These are the logs I get if I run the built flatpak with `--device=all --log-session-bus`: 
```
$ flatpak run --device=all --log-session-bus org.entangle_photo.Manager//test
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.entangle_photo.Manager
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.org.entangle_photo.Manager
C5: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
B10: <- org.freedesktop.DBus return from C9
*SKIPPED*
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 2)
*HIDDEN* (ping)
B11: <- org.freedesktop.DBus return from C10
B12: <- (no sender) return from C11
*REWRITTEN*
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 1)
B13: <- (no sender) return from C12
*REWRITTEN*
C13: -> org.gtk.vfs.Daemon call org.gtk.vfs.Daemon.ListMonitorImplementations at /org/gtk/vfs/Daemon
*HIDDEN* (ping)
B14: <- (no sender) return from C13
*REWRITTEN*
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.entangle_photo.Manager
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.org.entangle_photo.Manager
C5: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
B10: <- org.freedesktop.DBus return from C9
*SKIPPED*
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 2)
*HIDDEN* (ping)
B11: <- org.freedesktop.DBus return from C10
B12: <- (no sender) return from C11
*REWRITTEN*
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 1)
B13: <- (no sender) return from C12
*REWRITTEN*
C13: -> org.gtk.vfs.Daemon call org.gtk.vfs.MountTracker.ListMountableInfo at /org/gtk/vfs/mounttracker
*HIDDEN* (ping)
B14: <- (no sender) return from C13
*REWRITTEN*
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.entangle_photo.Manager
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.org.entangle_photo.Manager
C5: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
B10: <- org.freedesktop.DBus return from C9
*SKIPPED*
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.RequestName at /org/freedesktop/DBus
B11: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
B12: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B13: <- org.freedesktop.DBus return from C10
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.GetId at /org/freedesktop/DBus
B14: <- org.freedesktop.DBus return from C11
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C15: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
B15: <- org.freedesktop.DBus return from C12
B16: <- org.freedesktop.DBus return from C13
B17: <- org.freedesktop.DBus return from C14
B18: <- org.freedesktop.DBus return from C15
C16: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B19: <- org.freedesktop.DBus return from C16
C17: -> :1.72 call org.freedesktop.DBus.Properties.GetAll at /org/freedesktop/portal/desktop
B61722: <- :1.72 return from C17
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.entangle_photo.Manager
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.org.entangle_photo.Manager
C5: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.entangle_photo.Manager
C10: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
*SKIPPED*
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
B10: <- org.freedesktop.DBus return from C9
*SKIPPED*
B11: <- org.freedesktop.DBus return from C10
*SKIPPED*

(entangle:3): Gtk-WARNING **: 17:20:58.941: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
C18: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C19: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
C20: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
C11: -> org.freedesktop.portal.IBus call org.freedesktop.DBus.Peer.Ping at /org/freedesktop/IBus
B20: <- org.freedesktop.DBus return from C18
B21: <- org.freedesktop.DBus return from C19
B22: <- org.freedesktop.DBus return from C20
B39490: <- :1.14 return from C11
C21: -> all signal org.gtk.Actions.Changed at /org/entangle_photo/Manager
C12: -> org.freedesktop.portal.IBus call org.freedesktop.IBus.Portal.CreateInputContext at /org/freedesktop/IBus
B30369: <- :1.23 call org.gtk.Actions.DescribeAll at /org/entangle_photo/Manager
B30373: <- :1.23 call org.freedesktop.DBus.Properties.GetAll at /org/entangle_photo/Manager
C22: -> :1.23 return from B30369
C23: -> :1.23 return from B30373
B39491: <- :1.14 return from C12
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C15: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C16: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B12: <- org.freedesktop.DBus return from C13
B13: <- org.freedesktop.DBus return from C14
B14: <- org.freedesktop.DBus return from C15
B15: <- org.freedesktop.DBus return from C16
C17: -> :1.14 call org.freedesktop.DBus.Properties.GetAll at /org/freedesktop/IBus/InputContext_583
B39492: <- :1.14 return from C17
C18: -> :1.14 call org.freedesktop.DBus.Properties.Set at /org/freedesktop/IBus/InputContext_583
C19: -> :1.14 call org.freedesktop.IBus.InputContext.SetCapabilities at /org/freedesktop/IBus/InputContext_583
B39493: <- :1.14 return error org.freedesktop.DBus.Error.InvalidArgs from C18
*Unexpected reply*
B30376: <- :1.23 call org.gtk.Actions.DescribeAll at /org/entangle_photo/Manager/window/1
C24: -> :1.23 return from B30376
C20: -> org.freedesktop.portal.IBus call org.freedesktop.IBus.Portal.CreateInputContext at /org/freedesktop/IBus
B39494: <- :1.14 return from C20
C21: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C22: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C23: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B16: <- org.freedesktop.DBus return from C21
B17: <- org.freedesktop.DBus return from C22
B18: <- org.freedesktop.DBus return from C23
C24: -> :1.14 call org.freedesktop.DBus.Properties.GetAll at /org/freedesktop/IBus/InputContext_584
B39495: <- :1.14 return from C24
C25: -> :1.14 call org.freedesktop.DBus.Properties.Set at /org/freedesktop/IBus/InputContext_584
C26: -> :1.14 call org.freedesktop.IBus.InputContext.SetCapabilities at /org/freedesktop/IBus/InputContext_584
B39496: <- :1.14 return error org.freedesktop.DBus.Error.InvalidArgs from C25
*Unexpected reply*
C27: -> :1.14 call org.freedesktop.DBus.Properties.Set at /org/freedesktop/IBus/InputContext_584
C28: -> :1.14 call org.freedesktop.IBus.InputContext.FocusIn at /org/freedesktop/IBus/InputContext_584
B39497: <- :1.14 return from C27
*Unexpected reply*
C29: -> :1.14 call org.freedesktop.IBus.InputContext.SetCursorLocation at /org/freedesktop/IBus/InputContext_584
B39498: <- :1.14 signal org.freedesktop.IBus.InputContext.UpdatePreeditText at /org/freedesktop/IBus/InputContext_584
C30: -> :1.14 call org.freedesktop.IBus.InputContext.SetCursorLocation at /org/freedesktop/IBus/InputContext_584
C31: -> :1.14 call org.freedesktop.IBus.InputContext.FocusOut at /org/freedesktop/IBus/InputContext_584
B39499: <- :1.14 signal org.freedesktop.IBus.InputContext.UpdatePreeditText at /org/freedesktop/IBus/InputContext_584
B39500: <- :1.14 signal org.freedesktop.IBus.InputContext.UpdatePreeditText at /org/freedesktop/IBus/InputContext_584
B39501: <- :1.14 signal org.freedesktop.IBus.InputContext.UpdatePreeditText at /org/freedesktop/IBus/InputContext_584
B39502: <- :1.14 signal org.freedesktop.IBus.InputContext.UpdatePreeditText at /org/freedesktop/IBus/InputContext_584
B39503: <- :1.14 signal org.freedesktop.IBus.InputContext.UpdatePreeditText at /org/freedesktop/IBus/InputContext_584
C32: -> :1.14 call org.freedesktop.IBus.InputContext.Reset at /org/freedesktop/IBus/InputContext_584
C33: -> :1.14 call org.freedesktop.IBus.Service.Destroy at /org/freedesktop/IBus/InputContext_584
C34: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C35: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
B19: <- org.freedesktop.DBus return from C34
B20: <- org.freedesktop.DBus return from C35

(entangle:3): GLib-GObject-WARNING **: 17:21:08.181: invalid unclassed pointer in cast to 'GtkWidget'

(entangle:3): Gtk-CRITICAL **: 17:21:08.181: gtk_widget_destroy: assertion 'GTK_IS_WIDGET (widget)' failed
C25: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C26: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C27: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C28: -> org.freedesktop.DBus call org.freedesktop.DBus.ReleaseName at /org/freedesktop/DBus
B23: <- org.freedesktop.DBus return from C25
B24: <- org.freedesktop.DBus return from C26
B25: <- org.freedesktop.DBus return from C27
B21: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
B26: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameLost at /org/freedesktop/DBus
B21: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
B27: <- org.freedesktop.DBus return from C28
*Unexpected reply*
```

barthalion commented:
Have you also looked at `--log-system-bus`?

barthalion commented on entangle-2.8-appdata.patch:
```diff
@@ -0,0 +1,41 @@
+diff -Naur entangle-2.0/src/entangle.appdata.xml.in entangle-2.0/src/entangle.appdata.xml.in
```
Should be upstreamed if isn't already.

barthalion replied:
Never mind, just noticed it mentioned in the PR description… Sorry.

barthalion commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,128 @@
+                    "type": "archive",
+                    "url": "http://entangle-photo.org/download/sources/entangle-2.0.tar.xz",
+                    "sha256": "7eafc72374178b85c7c2f28a5afaebe702f8f8733ba2bfc401a575464f42f65f",
+                    "git-init" : true
```
What is it needed for?

manuq replied:
Good catch! Removed.

manuq commented:
> Have you also looked at `--log-system-bus`?

I got no output at all from `--log-system-bus`, so I removed it on purpose from my test command.

berrange commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,127 @@
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=x11",
+        "--socket=wayland"
```
IIUC, we'll need to add  "--device=all" so that the flatpak can access the USB device for the camera on the host.

berrange replied:
Also "--filesystem=xdg-pictures" so that it can read + write images.

It is possible we'll need --share=network in order to communicate with IP based cameras, though I'm not 100% sure on that one, as I'm not clear on network connectivity you get by default.

manuq replied:
> IIUC, we'll need to add "--device=all" so that the flatpak can access the USB device for the camera on the host.

Yeah, [is not ideal](http://docs.flatpak.org/en/latest/sandbox-permissions.html#device-access) but I guess for cameras we'll have to. I couldn't find a list of all the options that this receives to avoid passing "all". For now whlie testing I'm passing it by command line like this: `flatpak run --device=all ... `.

bilelmoussaoui replied:
You can't limit devices access for now

hfiguiere replied:
if you want to access the camera, you might want to have `libusb`. Look at Darktable flatpak, it works there.

berrange commented:
FYI if you want to debug why entangle fails to connect / find cameras, then try adding '--debug-entangle' and '--debug-gphoto' command line args when launching entangle. It will print a *huge* amount of logging information to stderr. If you can't see what's wrong from that, upload the logs somewhere I can see them & I'll take a look to see if I can spot a reasons.

berrange commented:
I understand from https://github.com/flathub/flathub/wiki/App-Submission that preferred practice is for the upstream app maintainer to have access to the flathub repository for their apps. I am the upstream maintainer for Entangle, but @manuq has been very helpfully putting in the time & effort to create the flatpak config for me.

bilelmoussaoui commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,127 @@
+    "sdk": "org.gnome.Sdk",
+    "command": "entangle",
+    "rename-icon": "entangle",
+    "rename-desktop-file": "entangle.desktop",
```
We would prefer if those renames happen upstream in the future. cc @berrange

manuq commented:
> FYI if you want to debug why entangle fails to connect / find cameras, then try adding '--debug-entangle' and '--debug-gphoto' command line args when launching entangle. It will print a _huge_ amount of logging information to stderr. If you can't see what's wrong from that, upload the logs somewhere I can see them & I'll take a look to see if I can spot a reasons.

Ah thanks for the debug instructions! This is what I was looking for.

manuq commented:
> I understand from https://github.com/flathub/flathub/wiki/App-Submission that preferred practice is for the upstream app maintainer to have access to the flathub repository for their apps. I am the upstream maintainer for Entangle, but @manuq has been very helpfully putting in the time & effort to create the flatpak config for me.

Yeah @berrange hopefully you can become the maintainer of this flathub repository and coordinate with upcoming releases, etc. My intention is to provide an initial working flatpak. I can also help you maintain it if you want.

Erick555 commented on org.entangle_photo.Manager.json:
```diff
@@ -0,0 +1,127 @@
+            ]
+        },
+        {
+            "name": "gudev",
```
@manuq  Use [gudev submodule](https://github.com/flathub/shared-modules/blob/master/gudev/gudev.json) instead.

LeandroStanger commented:
bot, build org.entangle_photo.Manager

berrange commented:
I've submitted a new PR that was derived from the work done on this PR, and it is now able to successfully connect & capture photos.

hfiguiere commented:
Someone with the permissions should close this PR.
